File: csharp/bottle-song/BottleSong.cs
using System.Collections.Generic;

public static class BottleSong
{
    public static IEnumerable<string> Recite(int startBottles, int takeDown)
    {
        var numberWords = new[] { "No", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten" };
        var bottleSong = new List<string>();
        for (var i = 0; i < takeDown; i++)
        {
            string beforeBottlesHanging = startBottles == 1 ? "bottle" : "bottles";
            string afterBottleHanging = startBottles == 2 ? "bottle" : "bottles";
            bottleSong.Add($"{numberWords[startBottles]} green {beforeBottlesHanging} hanging on the wall,");
            bottleSong.Add($"{numberWords[startBottles]} green {beforeBottlesHanging} hanging on the wall,");
            bottleSong.Add("And if one green bottle should accidentally fall,");
            bottleSong.Add($"There'll be {numberWords[startBottles - 1].ToLower()} green {afterBottleHanging} hanging on the wall.");
            startBottles--;
            if (i < takeDown - 1) bottleSong.Add("");
        }
        return bottleSong.ToArray();
    }
}

File: javascript/bottle-song/bottle-song.js
//
// This is only a SKELETON file for the 'Bottle Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const NUMBER_WORDS = [
  'no', 'one', 'two', 'three', 'four',
  'five', 'six', 'seven', 'eight', 'nine', 'ten'
]

export const recite = (initialBottlesCount, takeDownCount) => {
  let songLyrics = [];
  for (let i = 0; i < takeDownCount; i++) {
    const currentBottles = initialBottlesCount - i;
    songLyrics = songLyrics.concat(generateVerse(currentBottles));
    if (i < takeDownCount - 1) {
      songLyrics.push('');
    }
  }
  return songLyrics;
};

const generateVerse = (totalCount) => {
  const currentWordCap = NUMBER_WORDS[totalCount].charAt(0).toUpperCase() + NUMBER_WORDS[totalCount].slice(1);
  const nextWordLow = NUMBER_WORDS[totalCount - 1];

  const currentPlural = totalCount === 1 ? '' : 's';
  const nextPlural = totalCount - 1 === 1 ? '' : 's';

  return [
    `${currentWordCap} green bottle${currentPlural} hanging on the wall,`,
    `${currentWordCap} green bottle${currentPlural} hanging on the wall,`,
    `And if one green bottle should accidentally fall,`,
    `There'll be ${nextWordLow} green bottle${nextPlural} hanging on the wall.`
  ];
};